src/application/services/useEditor.ts

```diff
@@ -0,0 +1,114 @@
+import type { BlockTool } from '@editorjs/editorjs';
+import Editor, { type OutputData, type API } from '@editorjs/editorjs';
+// @ts-expect-error editor plugins have no types
+import Header from '@editorjs/header';
+import type { Ref } from 'vue';
+import { onBeforeUnmount, onMounted } from 'vue';
+import { useAppState } from './useAppState';
+import type EditorTool from '@/domain/entities/EditorTool';
+import { loadScript } from '@/infrastructure/utils/load-script';
+
+/**
+ * Downloaded tools data structure
+ */
+type DownloadedTools = Record<string, BlockTool>;
+
+/**
+ * UseEditor composable params
+ */
+interface UseEditorParams {
+  /**
+   * Host element id
+   */
+  id: string;
+
+  /**
+   * Editor initial content
+   */
+  content?: OutputData;
+
+  /**
+   * True if editor should not allow editing
+   */
+  isReadOnly?: boolean;
+
+  /**
+   * Handles content change in Editor
+   */
+  onChange?: (api: API) => void;
+}
+
+
+/**
+ * Handles Editor.js instance creation
+ *
+ * @param params - Editor.js params
+ */
+export function useEditor({ id, content, isReadOnly, onChange }: UseEditorParams): void {
+  /**
+   * Editor instance
+   */
+  let editor: Editor | undefined;
+
+  /**
+   * User notes tools
+   */
+  const { userEditorTools } = useAppState();
+
+  /**
+   * Download all the user tools and return a map to use in Editor.js constructor
+   *
+   * @param tools - tools data
+   */
+  async function downloadTools(tools: Ref<EditorTool[]>): Promise<DownloadedTools> {
+    const downloadedTools: DownloadedTools = {};
+
+    for (const tool of tools.value) {
+      if (tool.source.cdn === undefined) {
+        continue;
+      }
+
+      await loadScript(tool.source.cdn);
+
+      downloadedTools[tool.title] = window[tool.exportName as keyof typeof window];
```

Comment:
seems like there should be "name" instead of title

Reply:
title should be used of Toolbox title

Reply:
➕